### PR TITLE
Add list-ops exercise

### PR DESCRIPTION
**Pull request by ljagged**

### CODE REVIEW

File: exercises/list-ops/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+without using existing functions.
+
+## Running the tests
+
+To run the tests run the command `go test` from within the exercise directory.
+
+If the test suite contains benchmarks, you can run these with the `-bench`
+flag:
+
+    go test -bench .
```
Comment (leenipper) [inline exercises/list-ops/README.md]: See #1120 which was a recent change.  You can fetch and run configlet as mentioned in that PR to get this README.md updated with the "stock" changes.

File: exercises/list-ops/example.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	for idx, elt := range s {
+		newSlice[idx] = fn(elt)
+	}
+	return newSlice
+}
+
+// Reverse reverses the list
+func (s IntList) Reverse() IntList {
+	last := len(s) - 1
+	newSlice := make([]int, last+1)
```
Comment (leenipper) [inline exercises/list-ops/example.go]: Perhaps
```
newSlice := make([]int, len(s))
```

File: exercises/list-ops/listops_test.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	for _, tt := range foldTestCases {
+		if tt.property == "foldr" {
+			got = tt.list.Foldr(tt.fn, tt.initial)
+		} else {
+			got = tt.list.Foldl(tt.fn, tt.initial)
+		}
+		if got != tt.want {
+			t.Fatalf("Build for test case %q for property %s returned %d but was expected to return %d",
+				tt.name, tt.property, got, tt.want)
+		}
```
Comment (leenipper) [inline exercises/list-ops/listops_test.go]: Perhaps
```
} else {
  t.Logf("PASS: %s", tt.name)
}
```
to allow 'go test -v' to show more output.  This has been typical in most exercises.  Ditto for the other test funcs.

File: exercises/list-ops/listops_test.go [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+		want:     []int{1, 2, 3, 4, 5, 6},
+	},
+}
+
+func TestConcatMethod(t *testing.T) {
+	for _, tt := range concatTestCases {
+		got := tt.list.Concat(tt.args)
+		if !reflect.DeepEqual(tt.want, got) {
+			t.Fatalf("Build for test case %q for property %s returned %v but was expected to return %v",
+				tt.name, tt.property, got, tt.want)
```
Comment (leenipper) [inline exercises/list-ops/listops_test.go]: Consider using something along the lines of:
```
  t.Fatalf("FAIL: %q ... ", tt.name, ... )
```
Also, for brevity,  could use "expected: xxx  actual: yyy" pattern if you want. Survey some of the other exercises test code output for reference.

**Comment by ljagged** [comment]
Thanks for the suggestions. They were all very helpful. I've updated the code to make it more consistent with the other exercises.

**Comment by ljagged** [comment]
Thanks! It was fun!

**Pull request merged**